plugins/inputs/docker/docker.go

```diff
@@ -209,6 +211,27 @@ func (d *Docker) Gather(acc telegraf.Accumulator) error {
 	}
 	wg.Wait()
 
+	// Get disk usage data
+	if len(d.StorageObjects) > 0 {
+		objectTypes := []types.DiskUsageObject{}
+
+		for _, object := range d.StorageObjects {
+			if object == "container" {
+				objectTypes = append(objectTypes, types.ContainerObject)
+			} else if object == "image" {
+				objectTypes = append(objectTypes, types.ImageObject)
+			} else if object == "volume" {
+				objectTypes = append(objectTypes, types.VolumeObject)
+				// } else if object == "build-cache" {
+				// 	du_opts.Types = append(du_opts.Types, types.BuildCacheObject)
+			} else {
+				d.Log.Warnf("Unrecognized storage object type: %s", object)
+			}
+		}
+
+		d.gatherDiskUsage(acc, types.DiskUsageOptions{Types: objectTypes})
+	}
```

Comment:
How about
```suggestion
	if len(d.StorageObjects) > 0 {
		objectTypes := make([]types.DiskUsageObject, 0, len(d.StorageObjects))

		for _, object := range d.StorageObjects {
			switch object {
			case "container":
				objectTypes = append(objectTypes, types.ContainerObject)
			case "image":
				objectTypes = append(objectTypes, types.ImageObject)
			case "volume":
				objectTypes = append(objectTypes, types.VolumeObject)
			default:
				d.Log.Warnf("Unrecognized storage object type: %s", object)
			}
		}

		d.gatherDiskUsage(acc, types.DiskUsageOptions{Types: objectTypes})
	}
```

In any case you should check the options in the plugin's `Init()` function to make sure you don't spam the telegraf log-file on user-typos...